src/pyk/kcfg/kcfg.py

```diff
@@ -968,6 +969,24 @@ def prune(self, node_id: str) -> None:
         for node in nodes:
             self.remove_node(node.id)
 
+    def shortest_path_between(self, source_node_id: str, target_node_id: str) -> tuple[Successor, ...] | None:
+        paths = self.paths_between(source_node_id, target_node_id)
+        return sorted(paths, key=(lambda path: len(path)))[0]
+
+    def path_constraints(self, final_node_id: str) -> KInner:
+        path = self.shortest_path_between(self.get_unique_init().id, final_node_id)
+        if path is None:
+            raise ValueError(f'No path found to specified node: {final_node_id}')
+        curr_constraint: KInner = mlTop()
+        for edge in reversed(path):
+            if type(edge) is KCFG.Split:
+                assert len(edge.targets) == 1
+                _, csubst = edge.targets[0]
+                curr_constraint = mlAnd([csubst.constraint, curr_constraint])
```

Comment:
```suggestion
                _, csubst = edge.targets[0]
                curr_constraint = mlAnd([csubst.subst.ml_pred, csubst.constraint, curr_constraint])
```

The substitution here represents constarints on variables tat we should include in the path condition.

Reply:
I made the change suggested. The example I have been working with doesn't have anything in the subst field. Are substs sometimes generated on split edges?

Reply:
They can be, but not for IMP in this case. But in general yes.